seal_fhe/src/encryptor_decryptor.rs

```diff
@@ -35,10 +79,48 @@ use crate::{Ciphertext, Context, Plaintext, PublicKey, SecretKey};
  * "default NTT form". Decryption requires the input ciphertexts to be in the default
  * NTT form, and will throw an exception if this is not the case.
  */
-pub struct Encryptor {
+pub struct Encryptor<T = ()> {
     handle: *mut c_void,
+    _marker: PhantomData<T>,
+}
+
+/// An encryptor capable of symmetric encryptions.
+pub type SymmetricEncryptor = Encryptor<Sym>;
+
+/// An encryptor capable of asymmetric encryptions.
+pub type AsymmetricEncryptor = Encryptor<Asym>;
```

Comment:
Why not have a type for `Encrytor<SymAsym>`?.

Reply:
Honestly because I thought `SymmetricAsymmetricEncryptor` was too wordy :shrug: I only ended up needing to reference `Encryptor<SymAsym>` in a couple tests so I thought it was fine to not make a synonym for it. If you want one I can add it.

Reply:
Might be worth adding just for completeness.